File: CPP Unit 05/Static/Q4.cpp
Single File Programming Question
Problem Statement:



In a school competition, each Student object can have a friend (another Student object). The teacher wants to analyze pairs of friends by comparing their marks. Create a Student class that can point to another Student object (friend). 



For each student pair, calculate the absolute difference between their marks and determine who scored higher. If a student has no friend, simply indicate that condition.

Input format :
The first line of input consists of an integer representing the marks of student 1.

The second line consists of an integer (0 or 1) indicating whether student 1 has a friend (0 for no, 1 for yes).

If the second line is 1, the third line consists of an integer representing the marks of student 2.

Output format :
If the student has a friend:

The first line prints "Difference: " followed by the absolute difference between their marks
The second line prints "Student 1 higher", "Student 2 higher", or "Both equal" based on comparison
If the student has no friends:

The first line prints "No friend"


Refer to the sample output for the formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

Marks range from 0 to 100

The friend indicator is either 0 or 1

All inputs are valid integers

Sample test cases :
Input 1 :
60
1
40
Output 1 :
Difference: 20
Student 1 higher
Input 2 :
30
1
70
Output 2 :
Difference: 40
Student 2 higher
Input 3 :
45
1
45
Output 3 :
Difference: 0
Both equal
Input 4 :
100
0
100
Output 4 :
No friend


#include <iostream>
#include <cmath>
using namespace std;

class Student {
public:
    int marks;
    Student* frnd;

    Student(int m) {
        marks = m;
        frnd = nullptr;
    }
};

int main() {
    int m1;
    cin >> m1;

    int hasFriend;
    cin >> hasFriend;

    Student s1(m1);

    if (hasFriend == 0) {
        cout << "No friend";
        return 0;
    }

    int m2;
    cin >> m2;

    Student s2(m2);
    s1.frnd = &s2;

    int diff = abs(s1.marks - s1.frnd->marks);
    cout << "Difference: " << diff << endl;

    if (s1.marks > s1.frnd->marks) {
        cout << "Student 1 higher";
    } else if (s1.marks < s1.frnd->marks) {
        cout << "Student 2 higher";
    } else {
        cout << "Both equal";
    }

    return 0;
}

File: Java/Unit 01/Lec 08/Q4.java
Single File Programming Question
Problem Statement



Maya is given a two-digit number and needs to find the sum of its digits. She wants to know if the sum is less than 10 or not. Your task is to write a program to calculate and print the sum of the digits and provide a message based on the result.

Input format :
The input consists of an integer n.

Output format :
The output prints the sum of the digits and a message indicating whether it's less than 10 or not.

"Digit Sum: <<sum>>

The sum of the digits is less/not less than 10."

If n is not a two-digit number, print "Not a valid two-digit number."



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ n ≤ 1000

Sample test cases :
Input 1 :
12
Output 1 :
Digit Sum: 3
The sum of the digits is less than 10.
Input 2 :
29
Output 2 :
Digit Sum: 11
The sum of the digits is not less than 10.
Input 3 :
55
Output 3 :
Digit Sum: 10
The sum of the digits is not less than 10.
Input 4 :
345
Output 4 :
Not a valid two-digit number.



import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        if (n >= 10 && n <= 99) {
            int digit1 = n / 10;
            int digit2 = n % 10;
            int sum = digit1 + digit2;

            System.out.println("Digit Sum: " + sum);

            if (sum < 10) {
                System.out.println("The sum of the digits is less than 10.");
            } else {
                System.out.println("The sum of the digits is not less than 10.");
            }
        } else {
            System.out.print("Not a valid two-digit number.");
        }

        sc.close();
    }
}





  
